src/pages/layout.js

```diff
@@ -1,70 +1,42 @@
-import React, { useRef } from 'react'
+import React from 'react'
 
-import { TwoPanel, Sidebar, MainContent } from '@dhis2/d2-ui-core'
+import { TwoPanel, MainContent } from '@dhis2/d2-ui-core'
 import { Paper } from '@material-ui/core'
-import { Route, Link, Switch, HashRouter } from 'react-router-dom'
+import { Route, Switch, HashRouter } from 'react-router-dom'
 
 import AndroidSettingsContainer from '../components/android-settings-container'
 import menuSection from '../constants/menu-sections'
-import i18n from '@dhis2/d2-i18n'
 
 import { D2Shim } from '../utils/D2Shim'
 import layoutStyles from '../styles/Layout.module.css'
+import SideBar from '../components/sidebar'
 
 const styles = {
     twoPanelMain: {
         marginTop: '0rem',
     },
 }
 
-const Layout = props => {
-    const currentSection = useRef()
-    const sidebarRef = useRef()
-
-    const routeSections = menuSection
-    routeSections.push({
-        key: 'home',
-        path: '/',
-        component: <AndroidSettingsContainer />,
-    })
-
-    const changeSectionHandler = key => {
-        currentSection.current = key
-        if (key !== 'search' && sidebarRef.current) {
-            sidebarRef.current.clearSearchBox()
-        }
-    }
-
+const Layout = () => {
     return (
         <HashRouter>
             <TwoPanel mainStyle={styles.twoPanelMain}>
-                <div className={layoutStyles.paper__twoPanel__sideBar}>
-                    <Sidebar
-                        sections={menuSection.map(
-                            ({ key, label, path, icon }, i) => ({
-                                key,
-                                label,
-                                icon,
-                                containerElement: (
-                                    <Link to={path}> {label} </Link>
-                                ),
-                            })
-                        )}
-                        onChangeSection={changeSectionHandler}
-                        currentSection={props.currentSection}
-                        ref={sidebarRef}
-                    />
-                </div>
+                <SideBar />
                 <MainContent>
-                    <header>
-                        <h1 className={layoutStyles.paper__twoPanel__mainTitle}>
-                            {i18n.t('Android settings')}
-                        </h1>
-                    </header>
                     <Paper className={layoutStyles.paper__layout}>
                         <D2Shim>
                             <Switch>
-                                {routeSections.map(section => (
+                                <Route
+                                    path="/"
+                                    exact
+                                    render={() => (
+                                        <D2Shim>
```

Comment:
Curious if it is necessary to have another D2Shim when there is already one above on line 27?

Reply:
I tried to remove D2Shim from lines 33 and 44 and keep only the one from line 27 but it fails to get users and the amount of data that the user would sync.

I can remove it from line 27 and 33, and keep the one inside of the menuSection map (line 44).